### DISCUSSION TITLE
C++ graphics libraries

### DISCUSSION THREAD

**Discussion by actuallyroy** [community/community#35683] [Other Feature Feedback, Questions, & Ideas] [answered]
Someone tell me some good C++ graphics libraries

**Comment by SamNickGammer** [comment, marked as answer]
The best graphics library is OpenGL, which is cross-platform. But since that’s a very low level thing with C-style code, you’ll have to learn quite a lot.

However, there are many libraries that will take care of all the complicated shit for you, which I think is what you meant by graphics library.

For 2D, I’d recommend SFML by a mile. It has good documentation, support, easy to understand, comes with almost everything you need for a game (window management, audio as a few examples), and it’s generally modern C++ code.


`#include <SFML/Graphics.hpp> 
 
int main() 
{ 
    // create the window 
    sf::RenderWindow window(sf::VideoMode(800, 600), "My window"); 
 
    // run the program as long as the window is open 
    while (window.isOpen()) 
    { 
        // check all the window's events that were triggered since the last iteration of the loop 
        sf::Event event; 
        while (window.pollEvent(event)) 
        { 
            // "close requested" event: we close the window 
            if (event.type == sf::Event::Closed) 
                window.close(); 
        } 
 
        // clear the window with black color 
        window.clear(sf::Color::Black); 
 
        // draw everything here... 
        // window.draw(...); 
 
        // end the current frame 
        window.display(); 
    } 
 
    return 0; 
} `
